src/app/containers/RadioPageBlocks/Blocks/OnDemandRadio/index.jsx

```diff
@@ -0,0 +1,40 @@
+import React, { useContext } from 'react';
+import styled from 'styled-components';
+import { GEL_SPACING_QUAD } from '@bbc/gel-foundations/spacings';
+import { string } from 'prop-types';
+import { RequestContext } from '#contexts/RequestContext';
+import MediaMessage from '#pages/MediaAssetPage/MediaMessage';
+
+const MediaPlayerOuterWrapper = styled.div`
+  @media (min-width: 63rem) {
+    display: flex;
+    justify-content: center;
+  }
+`;
+
+const MediaPlayerInnerWrapper = styled.div`
+  flex-shrink: 0;
+  width: 50rem;
+  max-width: calc(100vw - ${GEL_SPACING_QUAD});
+`;
+
+const OnDemandRadioContainer = ({ id }) => {
+  const { platform } = useContext(RequestContext);
+  const isValidPlatform = ['amp', 'canonical'].includes(platform);
+
+  if (!isValidPlatform || !id) return null;
```

Comment:
Can we split this, for better readability?

```
if (!id) {
  return null;
}

if (!isValidPlatform) {
  return null;
}
```

Reply:
I think keeping the validation together like it currently is fine and for me it's not too hard to read. If it still concerns you then maybe some sort of abstraction like

```jsx
const validate = (id, platform) => (id && ['amp', 'canonical'].includes(platform))

const OnDemandRadioContainer = ({ id }) => {
  const { platform } = useContext(RequestContext);
  const isValid = validate(id, platform)

  if (!isValid) return null;
```

but maybe not necessary

Reply:
I'm happy to leave it as is - no action necessary